### PR TITLE
feat(platform-ui): added platform-ui repo to crayons

### PR DESCRIPTION
**Pull request by kishore-kumar-E3682**
## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] My commits have standard messages as mentioned in [Contributing Guidelines](./../blob/next/CONTRIBUTING.md)

### CODE REVIEW

File: packages/crayons-extended/platform-ui/fileutil.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+const prependFile = require('prepend-file');
```
Comment (arvindanta) [inline packages/crayons-extended/platform-ui/fileutil.js:1]: both packages has this fileutil.js. can we move these common things outside of both packages and import from one common place ?

File: packages/crayons-extended/platform-ui/package.json [hunk 2]
```diff
@@ -0,0 +1,62 @@
+{
+  "name": "@freshworks/platform-ui",
+  "version": "0.0.1",
+  "description": "Stencil Component Starter",
```
Comment (arvindanta) [inline packages/crayons-extended/platform-ui/package.json:4]: please change the description

File: packages/crayons-extended/platform-ui/package.json [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  "files": [
+    "dist/",
+    "loader/",
+    "react/"
+  ],
+  "scripts": {
+    "build": "npm run add:react  && npm run build:components && npm run build:react",
+    "build:components": "stencil build --docs",
+    "clean:react": "rimraf react",
+    "add:react": "npm run clean:react && npx mkdirp react",
```
Comment (arvindanta) [inline packages/crayons-extended/platform-ui/package.json:26]: please order the scripts in alphabetical order

File: packages/crayons-extended/platform-ui/readme.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
+![Built With Stencil](https://img.shields.io/badge/-Built%20With%20Stencil-16161d.svg?logo=data%3Aimage%2Fsvg%2Bxml%3Bbase64%2CPD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0idXRmLTgiPz4KPCEtLSBHZW5lcmF0b3I6IEFkb2JlIElsbHVzdHJhdG9yIDE5LjIuMSwgU1ZHIEV4cG9ydCBQbHVnLUluIC4gU1ZHIFZlcnNpb246IDYuMDAgQnVpbGQgMCkgIC0tPgo8c3ZnIHZlcnNpb249IjEuMSIgaWQ9IkxheWVyXzEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgeG1sbnM6eGxpbms9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkveGxpbmsiIHg9IjBweCIgeT0iMHB4IgoJIHZpZXdCb3g9IjAgMCA1MTIgNTEyIiBzdHlsZT0iZW5hYmxlLWJhY2tncm91bmQ6bmV3IDAgMCA1MTIgNTEyOyIgeG1sOnNwYWNlPSJwcmVzZXJ2ZSI%2BCjxzdHlsZSB0eXBlPSJ0ZXh0L2NzcyI%2BCgkuc3Qwe2ZpbGw6I0ZGRkZGRjt9Cjwvc3R5bGU%2BCjxwYXRoIGNsYXNzPSJzdDAiIGQ9Ik00MjQuNywzNzMuOWMwLDM3LjYtNTUuMSw2OC42LTkyLjcsNjguNkgxODAuNGMtMzcuOSwwLTkyLjctMzAuNy05Mi43LTY4LjZ2LTMuNmgzMzYuOVYzNzMuOXoiLz4KPHBhdGggY2xhc3M9InN0MCIgZD0iTTQyNC43LDI5Mi4xSDE4MC40Yy0zNy42LDAtOTIuNy0zMS05Mi43LTY4LjZ2LTMuNkgzMzJjMzcuNiwwLDkyLjcsMzEsOTIuNyw2OC42VjI5Mi4xeiIvPgo8cGF0aCBjbGFzcz0ic3QwIiBkPSJNNDI0LjcsMTQxLjdIODcuN3YtMy42YzAtMzcuNiw1NC44LTY4LjYsOTIuNy02OC42SDMzMmMzNy45LDAsOTIuNywzMC43LDkyLjcsNjguNlYxNDEuN3oiLz4KPC9zdmc%2BCg%3D%3D&colorA=16161d&style=flat-square)
```
Comment (arvindanta) [inline packages/crayons-extended/platform-ui/readme.md:1]: please change this readme and add appropiate contents

File: packages/crayons-extended/platform-ui/src/styles/_theme.scss [hunk 5]
```diff
@@ -0,0 +1,333 @@
+// Color/Theme related config variables
```
Comment (arvindanta) [inline packages/crayons-extended/platform-ui/src/styles/_theme.scss:1]: In the root already there is a folder named styles with most of these style present.

Can you reuse that folder and only add what is unique to this package here ?.

You can refer to how it is done in `crayons-core` package

**Review by arvindanta — CHANGES_REQUESTED** [review]

File: packages/crayons-extended/platform-ui/src/components/filter/readme.md [hunk 6]
```diff
@@ -0,0 +1,141 @@
+# fw-filter
+
+A Dynamic filter component that will generate a filter form based on the conditionSchema and filters.
```
Comment (arvindanta) [inline packages/crayons-extended/platform-ui/src/components/filter/readme.md:3]: Please add a working demo for all the platform-ui components.

**Review by arvindanta — CHANGES_REQUESTED** [review]

File: packages/crayons-extended/platform-ui/fileutil.js (hunk 1 above)
Comment (kishore-kumar-E3682) [reply packages/crayons-extended/platform-ui/fileutil.js]: Moved them to scripts folder at the root

File: packages/crayons-extended/platform-ui/package.json (hunk 2 above)
Comment (kishore-kumar-E3682) [reply packages/crayons-extended/platform-ui/package.json]: Fixed it.

File: packages/crayons-extended/platform-ui/src/styles/_theme.scss (hunk 5 above)
Comment (kishore-kumar-E3682) [reply packages/crayons-extended/platform-ui/src/styles/_theme.scss]: Reused the common CSS

File: packages/crayons-extended/platform-ui/readme.md (hunk 4 above)
Comment (kishore-kumar-E3682) [reply packages/crayons-extended/platform-ui/readme.md:1]: Updated the Readme

File: packages/crayons-extended/platform-ui/src/components/platform-table/readme.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
+# fw-platform-table
+
+## Sorting
+
+`fwSort` event will be triggered whenever the user selects the sort by or the order by option in the dropdown.
```
[the remark this answers is not in the document]
Comment (kishore-kumar-E3682) [reply packages/crayons-extended/platform-ui/src/components/platform-table/readme.md:7]: Added examples

File: packages/crayons-extended/platform-ui/src/components/filter/readme.md (hunk 6 above)
Comment (kishore-kumar-E3682) [reply packages/crayons-extended/platform-ui/src/components/filter/readme.md:3]: Added examples

File: packages/crayons-extended/platform-ui/package.json [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    "url": "https://github.com/ionic-team/stencil-component-starter.git"
+  },
+  "files": [
+    "dist/",
+    "loader/",
+    "react/"
+  ],
+  "scripts": {
+    "add:react": "npm run clean:react && npx mkdirp react",
+    "add:css": "npm run clean:css && npx mkdirp css",
```
Comment (arvindanta) [inline packages/crayons-extended/platform-ui/package.json:24]: add:css is the first script followed by add:react

File: packages/crayons-extended/platform-ui/package.json [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+  },
+  "dependencies": {
+    "@freshworks/crayons": "^3.4.2",
+    "@freshworks/crayons-i18n": "^3.1.1",
+    "@stencil/core": "2.9.0",
+    "stencil": "^0.0.5",
+    "uuid": "^8.3.2"
+  },
+  "devDependencies": {
+    "@types/jest": "^26.0.24",
```
Comment (arvindanta) [inline packages/crayons-extended/platform-ui/package.json:47]: please add `react-output-target` as dev deps. Please check crayons-core package.json

File: packages/crayons-extended/platform-ui/src/index.html [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+  <head>
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=5.0" />
+    <title>Stencil Component Starter</title>
+
+    <script type="module" src="/build/platform-ui.esm.js"></script>
+    <script nomodule src="/build/platform-ui.js"></script>
+  </head>
+  <body>
+    <my-component first="Stencil" last="'Don't call me a framework' JS"></my-component>
```
Comment (arvindanta) [inline packages/crayons-extended/platform-ui/src/index.html:12]: please remove this `my-component` . Maybe add some example of a component present in platformui package

File: packages/crayons-extended/platform-ui/src/utils/utils.spec.ts [hunk 11]
```diff
@@ -0,0 +1,21 @@
+import { format } from './utils';
+
+describe('format', () => {
```
Comment (arvindanta) [inline packages/crayons-extended/platform-ui/src/utils/utils.spec.ts:3]: please remove the unused test cases and add the test cases from crayons-core for the respective methods

File: packages/crayons-extended/platform-ui/src/utils/utils.ts [hunk 12]
```diff
@@ -0,0 +1,25 @@
+export function format(first: string, middle: string, last: string): string {
```
Comment (arvindanta) [inline packages/crayons-extended/platform-ui/src/utils/utils.ts:1]: please remove the `format` function

**Review by arvindanta — CHANGES_REQUESTED** [review]

File: packages/crayons-extended/platform-ui/src/utils/utils.ts (hunk 12 above)
Comment (kishore-kumar-E3682) [reply packages/crayons-extended/platform-ui/src/utils/utils.ts:1]: This function is used in the _plop-template_ hbs files. These are available in crayons-core too.

File: packages/crayons-extended/platform-ui/package.json (hunk 9 above)
Comment (kishore-kumar-E3682) [reply packages/crayons-extended/platform-ui/package.json:48]: Added the missing dev scripts

File: packages/crayons-extended/platform-ui/package.json (hunk 8 above)
Comment (kishore-kumar-E3682) [reply packages/crayons-extended/platform-ui/package.json:23]: Modified the order

**Review by arvindanta — APPROVED** [review]

**Pull request merged**